baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -127,6 +169,62 @@ def handle(self, message: kombu.Message) -> None:
             message.ack()
 
 
+class KombuBatchMessageHandler(MessageHandler):
+    def __init__(
+        self,
+        baseplate: Baseplate,
+        name: str,
+        handler_fn: BatchHandler,
+        error_handler_fn: Optional[BatchErrorHandler] = None,
+    ):
+        self.baseplate = baseplate
+        self.name = name
+        self.handler_fn = handler_fn
+        self.error_handler_fn = error_handler_fn
+
+    def handle(self, messages: Sequence[kombu.Message]) -> None:  # pylint: disable=arguments-differ
+        logger.info("Processing batch with %i messages", len(messages))
+
+        context = self.baseplate.make_context_object()
+        try:
+            # We place the call to ``baseplate.make_server_span`` inside the
+            # try/except block because we still want Baseplate to see and
+            # handle the error (publish it to error reporting)
+            with self.baseplate.make_server_span(context, self.name) as span:
+                span.set_tag("kind", "batch_consumer")
+                span.set_tag("size", len(messages))
+
+                message: kombu.Message
+                for i, message in enumerate(messages):
+                    delivery_info = message.delivery_info
+                    span.set_tag(f"amqp.routing_key_{i}", delivery_info.get("routing_key", ""))
+                    span.set_tag(f"amqp.consumer_tag_{i}", delivery_info.get("consumer_tag", ""))
+                    span.set_tag(f"amqp.delivery_tag_{i}", delivery_info.get("delivery_tag", ""))
+                    span.set_tag(f"amqp.exchange_{i}", delivery_info.get("exchange", ""))
```

Comment:
I'm still not sure that we want to do this, this is going to create a lot of tags (4 per message in the batch).  I would still suggest using [the example I gave in a previous comment](https://github.com/reddit/baseplate.py/pull/483#discussion_r485097887) to only have unique sets of tags, or just leave these off altogether.

Reply:
That's a valid concern. The disadvantage of this would be that we don't know which messages have which tags.

Alternatively, would it be useful to have a single comma-delimited list of tags (for all items or uniques)?

Reply:
I think tags are useful for their raw text content in the trace view and they're useful for dimensioning like in tagged metrics. We should probably ignore the second use case here since it'd be hard to make meaningful dimensions in metrics out of something like this because the values are all muddled together.  Having large batch sizes mean massive trace payloads is a potential subtle and confusing bottleneck. So, for traces we probably just want to optimize for human consumption. De-duping and plopping into a comma-delimited list sounds great. I bet Andrew's right that in many cases it'll end up with single-item lists anyway.